File: packages/protocol-parser/src/schemaIndexToAbiType.ts
import { SchemaAbiType } from "@latticexyz/schema-type";
import { Schema } from "./common";

export function schemaIndexToAbiType(schema: Schema, schemaIndex: number): SchemaAbiType {
  if (schemaIndex < schema.staticFields.length) {
    return schema.staticFields[schemaIndex];
  }
  return schema.dynamicFields[schemaIndex - schema.staticFields.length];
}

File: packages/services/pkg/protocol-parser/schemaIndexToAbiType.go
package protocolparser

import schematype "latticexyz/mud/packages/services/pkg/schema-type"

func SchemaIndexToAbiType(schema Schema, schemaIndex int) schematype.SchemaType {
	if schemaIndex < len(schema.StaticFields) {
		return schema.StaticFields[schemaIndex]
	}
	return schema.DynamicFields[schemaIndex-len(schema.StaticFields)]
}
